### PR TITLE
refactor CI chaos tests: use matrix

### PR DESCRIPTION
**Pull request by rynewang**
Changes the CI chaos tests to use buildkite matrix feature. This allows us to reuse the environment set up code and decouple workload from injected chaos. Now you can define a new workload and get multiple different chaos for free.

One regression: previously we had the buildkite emojis for workloads (e.g. serve) and for chaos. Now we use the matrix cases to denote file names so no emojis.

### CODE REVIEW

File: .buildkite/pipeline.build_chaos.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        kind delete cluster
+      }
+      trap cleanup EXIT
+    - echo "--- Preparing chaos testing environment"
+    - ./python/ray/tests/chaos/prepare_env.sh
+    # First run: no faults
+    - echo "--- Running the workload without faults"
+    - "./python/ray/tests/chaos/{{matrix.workload}}.sh"
+    # Second run: with faults
+    - kubectl apply -f python/ray/tests/chaos/{{matrix.fault}}.yaml
```
Comment (rkooo567) [inline .buildkite/pipeline.build_chaos.yml:19]: is it possible to configure num_cpus for these workloads?

**Review by rkooo567 — COMMENTED** [review]
new lines in several files!

File: .buildkite/pipeline.build_chaos.yml (hunk 1 above)
Comment (rynewang) [reply .buildkite/pipeline.build_chaos.yml:19]: what do you mean by num_cpus? You mean the ray tasks' resource needs?

Comment (rkooo567) [reply .buildkite/pipeline.build_chaos.yml:19]: It'd be great if we can configure num cpus  & num workers from the k8s cluster.

**Review by rkooo567 — APPROVED** [review]
merge conflict

**Review by rkooo567 — APPROVED** [review]

**Comment by rkooo567** [comment]
merge conflict + update the description

**Comment by rkooo567** [comment]
@rynewang can you address the issue^? I'd like to merge my PR that requires this

**Comment by rynewang** [comment]
Conflict solved, description added

**Comment by rkooo567** [comment]
Updated a branch due to lots of premerge test failures

**Comment by rkooo567** [comment]
Seems like there's another merge conflict...

**Comment by rynewang** [comment]
merged master again

**Pull request merged**